hardware/Netatmo.cpp

```diff
@@ -504,8 +674,167 @@ uint64_t CNetatmo::UpdateValueInt(int HardwareID, const char* ID, unsigned char
+//			HardwareID, 
+//			deviceID, 
+//			sDeviceID.c_str(), 
+//			unit, 
+//			devType, 
+//			subType, 
+//			signallevel, 
+//			batterylevel, 
+//			nValue, 
+//			sValue, 
+//			devname.c_str(), 
+//			bUseOnOffAction, 
+//			user.c_str());
+//
+//		Log(LOG_STATUS,"Record added/updated with DeviceRowIdx = %d", (int)DeviceRowIdx);
+
+//		Log(LOG_STATUS, "UpdateValueInt: DeviceRowIdx=%d, ID=%s (%s), unit=%d, devType=%d, subType=%d, devname=%s", 
+//			(int)DeviceRowIdx,
+//			deviceID, 
+//			sDeviceID.c_str(), 
+//			unit, 
+//			devType, 
+//			subType, 
+//			devname.c_str());
+
+		std::vector<std::vector<std::string> > result;
+		std::string searchString = sDeviceID.substr(sDeviceID.length()-2,2);
+
+		result = m_sql.safe_query(
+			"SELECT ID, DeviceID, Name, type, subtype, unit, used, LastUpdate "\
+			"FROM DeviceStatus "\
+			"WHERE (HardwareID=%d "\
+				"AND ID <> %" PRIu64 " "\
+				"AND substr(iif("\
+					"deviceID==cast(deviceId as decimal) AND (length(deviceID)!=2 OR length(deviceID)!=4 OR length(deviceID)!=8) ,"\
+					"printf('%s',deviceID),"\
+					"upper(deviceID)), -2,2) =='%q' "\
+				"AND Type=%d) "\
+			"ORDER BY ID DESC",
+			m_HwdID, 
+			DeviceRowIdx,
+			"%X", 
+			searchString.c_str(), 
+			devType);
+		if (!result.empty()) {
+//			Log(LOG_STATUS, "Possible conversion found: %d matches found for %s (0x%s):",
+//				result.size(), 
+//				deviceID, 
+//				sDeviceID.c_str());
+			for (const auto& device : result)
+			{
+				if (!device.empty()) {
+					uint64_t  oldID = getValueFromString(device[0]);
+					std::string  oldDeviceID = device[1];
+					std::transform(oldDeviceID.begin(), oldDeviceID.end(), oldDeviceID.begin(), ::toupper);
+					unsigned char oldDevType = getValueFromString(device[3]);
+					unsigned char oldSubType = getValueFromString(device[4]);
+					unsigned char oldUnit = getValueFromString(device[5]);
+					unsigned char oldUsed = getValueFromString(device[6]);
+					std::string oldLastUpdate = device[7];
+
+					if (oldID == DeviceRowIdx)
+					{
+						Log(LOG_STATUS, "Case 0 (New device): Device: ID=%" PRIu64 ", DeviceID=0x%s, Name=%s, Type=%d, SubType=%d, unit=%d, used=%d, lastUpdate=%s",
+							oldID,
+							oldDeviceID.c_str(),
+							device[2].c_str(),
+							oldDevType,
+							oldSubType,
+							oldUnit,
+							oldUsed,
+							oldLastUpdate.c_str()
+						);
+					}
+					else if (oldUnit == unit && oldDevType == devType && oldSubType == subType)
+					{
+						Log(LOG_STATUS, "Case 1 (full match): Device: ID=%" PRIu64 ", DeviceID=0x%s, Name=%s, Type=%d, SubType=%d, unit=%d, used=%d, lastUpdate=%s",
+							oldID,
+							oldDeviceID.c_str(),
+							device[2].c_str(),
+							oldDevType,
+							oldSubType,
+							oldUnit,
+							oldUsed,
+							oldLastUpdate.c_str()
+						);
+						MergeDevices(oldID, DeviceRowIdx);
+						break; 	//One device per cycle
+					}
+					else if (oldUnit == unit && oldDevType == 82 && oldDevType == devType && oldSubType == 5 && subType == 113)
```

Comment:
As these if/then/else lines can become quite large, I use another coding formatting throughout the Domoticz code like
```

    else if (
        (oldUnit == unit)
        && (oldDevType == 82)
        && (oldDevType == devType)
        && (oldSubType == 5)
        && (subType == 113)
        )
    {
        ....
    }

```

I think this also looks a lot cleaner

And instead of these numbers, please use the real variable/defines, like defined in **hardwaretypes.h** or **RFXtrx.h**

82 -> pTypeTEMP_HUM

Numbers 'can' change, and nobody knows whats happening by looking at just numbers

Reply:
OK, but I looked in hardwaretypes.h, but didn't find matching names for the numbers. Maybe in RFXtrx.h; I'll take a look.

Reply:
Done (both requests)